src/index.ts

```diff
@@ -1856,15 +1861,27 @@ export class GossipSub extends EventEmitter<GossipsubEvents> implements PubSub<G
 
     this.mesh.set(topic, toAdd)
 
-    toAdd.forEach(async (id) => {
+    toAdd.forEach((id) => {
       this.log('JOIN: Add mesh link to %s in %s', id, topic)
-      await this.sendGraft(id, topic)
+      this.sendGraft(id, topic)
 
       // rust-libp2p
       // - peer_score.graft()
       // - Self::control_pool_add()
       // - peer_added_to_mesh()
     })
+
+    if (this.opts?.taggingEnabled ?? false) {
+      Array.from(toAdd).map(async (id) => {
```

Comment:
forEach or for-of

Reply:
probably for-of is better, then no need to create an array

Reply:
then I would have to make this whole method `async` and the motivation was to move away from unnecessary promises. i.e. promises would only be created for consumers who enable tagging.

Reply:
What if `peerStore.merge` throws?

Reply:
This should be refactored to a for-of

Reply:
> then I would have to make this whole method `async` and the motivation was to move away from unnecessary promises. i.e. promises would only be created for consumers who enable tagging.